src/lib/network_pool/transaction_pool.ml

```diff
@@ -1066,21 +1091,8 @@ struct
                           in
                           match add_res with
                           | Ok (verified, pool', dropped) ->
-                              ( if is_sender_local then
-                                let batch_num =
-                                  if t.remaining_in_batch > 0 then (
-                                    t.remaining_in_batch
-                                    <- t.remaining_in_batch - 1 ;
-                                    t.current_batch )
-                                  else (
-                                    t.remaining_in_batch
-                                    <- max_per_15_seconds - 1 ;
-                                    t.current_batch <- t.current_batch + 1 ;
-                                    t.current_batch )
-                                in
-                                Hashtbl.add_exn t.locally_generated_uncommitted
-                                  ~key:verified
-                                  ~data:(Time.now (), `Batch batch_num) ) ;
+                              if is_sender_local then
```

Comment:
Should we even try to add if a locally generated command already exists? We hit the replace condition if the transaction already exists and if it is the first nonce in the pool then dependent transactions are removed. https://github.com/MinaProtocol/mina/blob/74793a26dabd381bc65d6be767fb7559e8dadf48/src/lib/network_pool/indexed_pool.ml#L889
 If it is not the first nonce, then there is an assertion failure https://github.com/MinaProtocol/mina/blob/74793a26dabd381bc65d6be767fb7559e8dadf48/src/lib/network_pool/indexed_pool.ml#L854

Reply:
I think we can skip re-adding locally generated txns and just update the rebroadcast timer. We may want to update the rebroadcast timer of the dependent transactions as well.

Reply:
I suppose we'll also want something about the transaction to change so that replacing a transaction works as expected- Maybe change `fee > replace fee` instead of `>=` to be eligible for replace?
https://github.com/MinaProtocol/mina/blob/74793a26dabd381bc65d6be767fb7559e8dadf48/src/lib/network_pool/indexed_pool.ml#L881

Reply:
Ah sorry, there is another duplicate check that I did not see. So we probably don't need to handle the replace case but might want to update the timer and not error out with `Duplicate` ?

Reply:
> Should we even try to add if a locally generated command already exists? We hit the replace condition if the transaction already exists and if it is the first nonce in the pool then dependent transactions are removed. 

And then they're re-added (at least, as far as fees will allow) shortly after, so this should be fine.
https://github.com/MinaProtocol/mina/blob/74793a26dabd381bc65d6be767fb7559e8dadf48/src/lib/network_pool/indexed_pool.ml#L913

> Ah sorry, there is another duplicate check that I did not see. So we probably don't need to handle the replace case but might want to update the timer and log a message that timer was updated in the `Duplicate` case?

Great catch, thank you! I've updated to move the logic to the `Duplicate`.